docs/reference/cluster/stats.asciidoc

```diff
@@ -1307,6 +1307,125 @@ Number of selected nodes using the distribution flavor and file type.
 =====
 ====
 
+`snapshots`::
+(object)
+Contains statistics about the <<snapshot-restore,snapshot>> activity in the cluster.
++
+.Properties of `snapshots`
+[%collapsible%open]
+=====
+
+`current_counts`:::
+(object)
+Contains statistics which report the numbers of various ongoing snapshot activities in the cluster.
++
+.Properties of `current_counts`
+[%collapsible%open]
+======
+`snapshots`:::
+(integer)
+The total number of snapshots and clones currently being created by the cluster.
+
+`shard_snapshots`:::
+(integer)
+The total number of outstanding shard snapshots in the cluster.
+
+`snapshot_deletions`:::
+(integer)
+The total number of snapshot deletion operations that the cluster is currently running.
+
+`concurrent_operations`:::
+(integer)
+The total number of snapshot operations that the cluster is currently running concurrently. This is the total of the `snapshots` and `snapshot_deletions` entries, and
+is limited by <<snapshot-max-concurrent-ops,the `snapshot.max_concurrent_operations` setting>>.
+
+`cleanups`:::
+(integer)
+The total number of repository cleanup operations that the cluster is currently running. These operations do not count towards the total
+number of concurrent operations.
+======
+
+
+`repositories`:::
+(object)
+Contains statistics which report the progress of snapshot activities broken down by repository. This object contains one entry for each repository registered with the cluster.
++
+.Properties of `repositories`
+[%collapsible%open]
+======
+
+`current_counts`:::
+(object)
+Contains statistics which report the numbers of various ongoing snapshot activities for this repository.
++
+.Properties of `current_counts`
+[%collapsible%open]
+=======
+`snapshots`:::
+(integer)
+The total number of snapshots currently running in this repository.
+
+`clones`:::
+(integer)
+The total number of snapshot clones currently running in this repository.
+
+`finalizations`:::
+(integer)
+The total number of snapshots and clone operations which are mostly complete except for the last "finalization" step.
+
+`deletions`:::
+(integer)
+The total number of snapshot deletion operations currently in this repository.
+
+`snapshot_deletions`:::
+(integer)
+The total number of snapshots that are currently being deleted from this repository.
+
+`active_deletions`:::
+(integer)
+The total number of snapshot deletion operations currently active in this repository.
```

Comment:
:) I was hoping nobody would ask. This starts to get into the details of how the snapshot state machine works, which isn't documented today. I think this number might be useful for debugging but I'd rather not have to describe the implementation deeply enough to explain it.

TLDR we don't delete snapshots and create snapshots concurrently, and while we're busy creating snapshots this will be `0` even if there are deletes waiting to run.

Reply:
I guess you can add `(useful for debugging)` just so as to avoid questions :D